### PR TITLE
Create marketplace in pan.dev

### PR DESCRIPTION
**Pull request by reutshal**
## Status
In Progress

## Description
- Created the marketplace tab in pan.dev
- One can create the marketplace-docs running npm run marketplace-docs
- Still need to add information to the documentation section 

@sserrata FYI

### CODE REVIEW

**Comment by glicht** [comment]
@reutshal please review lgtm and guardrails alerts

Also, please merge from master. I've put in a small perf improvement to the build at #585

**Comment by reutshal** [comment]
@yaakovi @Itay4 @glicht Please review

File: README.md [hunk 1]
```diff
@@ -58,6 +58,28 @@ Additional env vars that affect the generation of the docs:
...
+To generate the docs we use [pipenv](https://github.com/pypa/pipenv). Make sure to install pipenv by running: `pip3 install pipenv`.
+
+When working locally you can generate the reference docs by running:
+```
+npm run marketplace-docs
+```
+This task will download the packs information from the google cloud bucket and generate the packs docs for the marketplace tab. The generated docs are ignored by `.gitignore` and shouldn't be checked in as they are generated during the build.
+
+Additional env vars that affect the generation of the docs:
+* `FILE_REGEX`: specify a regex to select which docs to generate docs for. For example:
```
Comment (Itay4) [inline README.md:75]: is `FILE_REGEX` relevant?

File: content-repo/create-docs.sh [hunk 2]
```diff
@@ -139,6 +139,12 @@ if [[ ( "$PULL_REQUEST" == "true" || -n "$CI_PULL_REQUEST" ) && "$CONTENT_BRANCH
         echo "MAX_FILES set to: $MAX_FILES"
         export MAX_FILES
     fi
+
+    echo "$DIFF_FILES" | grep -v -E '^docs/|^content-repo/extra-docs/|^static/|^sidebars.js' || MAX_PACKS=20
+    if [ -n "MAX_PACKS" ]; then
+        echo "MAX_PACKS set to: MAX_PACKS"
```
Comment (Itay4) [inline content-repo/create-docs.sh:145]:
```suggestion
        echo "MAX_PACKS set to: $MAX_PACKS"
```

File: docusaurus.config.js [hunk 3]
```diff
@@ -38,81 +39,89 @@ module.exports = {
     algolia: {
       apiKey: "f5dfbee43cfa4c5024b10045c6d91461",
       indexName: "demisto",
-      algoliaOptions: { typoTolerance: false, hitsPerPage: 1000, filters: 'type:lvl1 OR type:content' } // Optional, if provided by Algolia
+      algoliaOptions: {
```
Comment (Itay4) [inline docusaurus.config.js:42]: im not big fan of changing things just for prettifying as i prefer to preserve the git history

File: genMarketplace.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    function readFileCallback(err) {
+      if (err) {
+        console.log(err);
+      }
+    }
+  );
+
+  marketplace.map((pack) => {
+    if (pack.contentItems) {
+      var FixedContentItems = {};
```
Comment (Itay4) [inline genMarketplace.js:139]: lets stay consistent with `const` and `let` and not use `var`

File: plopfile.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      },
+      {
+        type: "list",
+        name: "contentItems",
+        message: "Pack content items",
+      },
+      {
+        type: "input",
+        name: "changeLog",
+        message: "Pack change log",
```
Comment (Itay4) [inline plopfile.js:83]:
```suggestion
        message: "Pack changelog",
```

File: src/pages/marketplace/index.js [hunk 6]
```diff
@@ -0,0 +1,611 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (Itay4) [inline src/pages/marketplace/index.js:2]: whats this copyright for?

**Comment by glicht** [comment]
@sserrata there is a styling change in MD tables example:

Preview site:

image

Production site:

image

**Comment by glicht** [comment]
@sserrata also top NAV bar is not sticky on top. It scrolls up with the page. 

Preview site:

image

Production:

image

File: docusaurus.config.js (hunk 3 above)
Comment (sserrata) [reply docusaurus.config.js:42]: I agree, although the prettified result is much easier to diff than the original. I'll work on reverting all formatting changes to `docusaurus.config.js`.

File: src/pages/marketplace/index.js (hunk 6 above)
Comment (sserrata) [reply src/pages/marketplace/index.js:2]: This will be changed to reflect the same Palo Alto Networks copyright used throughout the production site.

**Comment by sserrata** [comment]
@glicht  I will introduce a change to preserve table formatting for the rest of the site. I'll also investigate why the top navbar isn't sticky anymore.

File: README.md (hunk 1 above)
Comment (reutshal) [reply README.md]: Thought about implementing it per pack, I will leave it tp phase two and remove for now

**Review by glicht — APPROVED** [review]

**Comment by glicht** [comment]
@sserrata @reutshal  seeing a page crash when doing a search two times in a row. I think it only happens when there is a history. I don't see this on other PRs. See video:

https://user-images.githubusercontent.com/1395797/115434895-e4b2d100-a211-11eb-89c3-cda555d6d3a6.mov

In the devtools console I see: 
```
TypeError: Cannot read property 'hierarchy' of undefined
    at 2428.a4770b19.js:1
    at Array.map (<anonymous>)
    at Te (2428.a4770b19.js:1)
    at Kr (main.22c434aa.js:2)
    at Ss (main.22c434aa.js:2)
    at yc (main.22c434aa.js:2)
    at fc (main.22c434aa.js:2)
    at sc (main.22c434aa.js:2)
    at main.22c434aa.js:2
    at t.unstable_runWithPriority (main.22c434aa.js:2)
is @ main.22c434aa.js:2
n.callback @ main.22c434aa.js:2
fr @ main.22c434aa.js:2
ds @ main.22c434aa.js:2
wc @ main.22c434aa.js:2
t.unstable_runWithPriority @ main.22c434aa.js:2
ji @ main.22c434aa.js:2
vc @ main.22c434aa.js:2
sc @ main.22c434aa.js:2
(anonymous) @ main.22c434aa.js:2
t.unstable_runWithPriority @ main.22c434aa.js:2
ji @ main.22c434aa.js:2
Ji @ main.22c434aa.js:2
Qi @ main.22c434aa.js:2
nc @ main.22c434aa.js:2
ko @ main.22c434aa.js:2
onStateChange @ 2428.a4770b19.js:1
onStateChange @ 2428.a4770b19.js:1
(anonymous) @ 2428.a4770b19.js:1
dispatch @ 2428.a4770b19.js:1
setStatus @ 2428.a4770b19.js:1
(anonymous) @ 2428.a4770b19.js:1
Promise.then (async)
(anonymous) @ 2428.a4770b19.js:1
Promise.then (async)
A @ 2428.a4770b19.js:1
onChange @ 2428.a4770b19.js:1
s @ main.22c434aa.js:2
u @ main.22c434aa.js:2
(anonymous) @ main.22c434aa.js:2
y @ main.22c434aa.js:2
st @ main.22c434aa.js:2
rt @ main.22c434aa.js:2
ct @ main.22c434aa.js:2
ht @ main.22c434aa.js:2
_ @ main.22c434aa.js:2
U @ main.22c434aa.js:2
$t @ main.22c434aa.js:2
Yt @ main.22c434aa.js:2
t.unstable_runWithPriority @ main.22c434aa.js:2
ji @ main.22c434aa.js:2
F @ main.22c434aa.js:2
Jt @ main.22c434aa.js:2
2428.a4770b19.js:1 Uncaught (in promise) TypeError: Cannot read property 'hierarchy' of undefined
    at 2428.a4770b19.js:1
    at Array.map (<anonymous>)
    at Te (2428.a4770b19.js:1)
    at Kr (main.22c434aa.js:2)
    at Ss (main.22c434aa.js:2)
    at yc (main.22c434aa.js:2)
    at fc (main.22c434aa.js:2)
    at sc (main.22c434aa.js:2)
    at main.22c434aa.js:2
    at t.unstable_runWithPriority (main.22c434aa.js:2)
```

**Comment by glicht** [comment]
> This pull request **fixes 1 alert** when merging [5442bc0](https://github.com/demisto/content-docs/commit/5442bc0cbd1fb2568d10758d1b7adcd070c4fb08) into [b36e0c3](https://github.com/demisto/content-docs/commit/b36e0c3756a72e2beeafcecfb4cf1faa940f8eb1) - [view on LGTM.com](https://lgtm.com/projects/g/demisto/content-docs/rev/pr-3a6abe1bc3f7096faa8d5b0a32279bcf00e88c88)
> 
> **fixed alerts:**
> 
> * 1 for Unused variable, import, function or class

@reutshal please take a look

**Comment by sserrata** [comment]
@glicht @reutshal Please see https://github.com/demisto/content-docs/pull/609 concerning the `TypeError: Cannot read property 'hierarchy' of undefined` issue. The fix was to pin to an earlier version of the algolia package. Please note that our site(s) using `yarn` did not encounter this issue, so it seems to be related to how `npm` handles dependencies.

**Comment by Itay4** [comment]
i think we should improve the table on the right and also dont display the `WORKS WITH THE FOLLOWING INTEGRATIONS:` if there are no integrations (as done in the platform)
image

**Comment by glicht** [comment]
> i think we should improve the table on the right and also dont display the `WORKS WITH THE FOLLOWING INTEGRATIONS:` if there are no integrations (as done in the platform)
> image

@sserrata @reutshal I think the table change was introduced as a result of the MD table support. I see in the following preview that it looks good: https://607eac154234f90f7e2ebdfe--demisto-content-docs.netlify.app/marketplace/details/ANYRUN . Let fix this one.

@Itay4 the issue with the empty "works with" we can do in a separate effort.

**Comment by reutshal** [comment]
@glicht @Itay4 Please take a final look

**Pull request merged**